contracts/contracts/harvest/BaseHarvester.sol

```diff
@@ -58,139 +97,287 @@ abstract contract BaseHarvester is Governable {
     /**
      * Address receiving rewards proceeds. Initially the Vault contract later will possibly
      * be replaced by another contract that eases out rewards distribution.
-     */
+     **/
     address public rewardProceedsAddress;
 
     /**
-     * @dev Constructor to set up initial internal state
-     * @param _vaultAddress Address of the Vault
-     */
-    constructor(address _vaultAddress) {
-        require(address(_vaultAddress) != address(0));
+     * All tokens are swapped to this token before it gets transferred
+     * to the `rewardProceedsAddress`. USDT for OUSD and WETH for OETH.
+     **/
+    address public immutable baseTokenAddress;
+    // Cached decimals for `baseTokenAddress`
+    uint256 public immutable baseTokenDecimals;
+
+    // Uniswap V2 path for reward tokens using Uniswap V2 Router
+    mapping(address => address[]) public uniswapV2Path;
+    // Uniswap V3 path for reward tokens using Uniswap V3 Router
+    mapping(address => bytes) public uniswapV3Path;
+    // Pool ID to use for reward tokens on Balancer
+    mapping(address => bytes32) public balancerPoolId;
+    // Packed indices of assets on the Curve pool
+    mapping(address => uint256) public curvePoolIndices;
```

Comment:
We could avoid packing, unpacking, and casting these in the code, by just using a struct with two 128 sized members? The rest would then be handled by the complier.

Would also document what's happening.

Reply:
Have changed the code to use a struct now